draft-ietf-quic-multipath.md

```diff
@@ -318,8 +318,11 @@ Path ID 0.
 
 After the handshake concluded if negotiation of multipath support succeeded,
 endpoints SHOULD use PATH_ACK frames instead of ACK frames,
-also for acknowledging so far unacknowledged 0-RTT packets, using
-Path ID 0. Similarly after a successful handshake, endpoints SHOULD also use
+including for acknowledging so far unacknowledged 0-RTT packets using Path ID 0.
+However, even after successful negotiation, endpoints MAY use ACK frames
+to acknowledge 0-RTT packets or 1-RTT packets received on initial path with Path ID 0,
+these ACK frames need to be processed correctly by peer.
```

Comment:
```suggestion
However, even after successful negotiation, endpoints can use ACK frames
to acknowledge 0-RTT packets or 1-RTT packets received on initial path with Path ID 0,
and these ACK frames need to be processed correctly by the peer.
```

Reply:
I think that we should keep the normative MAY. You want to say something like "MAY but we don't like it", but there is no way to say that in a standard. Either it is a protocol violation or it isn't. If it isn't, then the peer MAY do it.

Reply:
This MAY is not needed because we already have the SHOULD in the sentence before. I don't think we should say the same things twice normatively.

Reply:
That's one opinion. I have witnessed several IESG reviews asking "why do you say SHOULD and not MUST? If you say SHOULD, you must also have a clause explaining when the implementation MAY do something else." 

That construct is used at least once in RFC 9000, in section 11:

_The most appropriate error code ([Section 20](https://www.rfc-editor.org/rfc/rfc9000.html#error-codes)) SHOULD be included in the frame that signals the error. Where this specification identifies error conditions, it also identifies the error code that is used; though these are worded as requirements, different implementation strategies might lead to different errors being reported. In particular, an endpoint MAY use any applicable error code when it detects an error condition; a generic error code (such as PROTOCOL_VIOLATION or INTERNAL_ERROR) can always be used in place of specific error codes._

On the other hand, I don't care all that much...

Reply:
The point of adding this sentence is to make the receiver side aware and not to specify any requirements for the sender. There the sentence is rather stating a fact (like "note, this can happen and is valid") that setting a normative protocol requirement. If we want to phrase a requirement here, it should be on the receiver (as Quentin also says below). So basically saying something like:

`Endpoints MUST process ACK frames that acknowledge 0-RTT packets or 1-RTT packets for Path ID 0, even after successful negotiation of this extension.`

Further the intention of adding something here as discussed in the issue was to give a bit of a reasoning why there is a SHOULD in the above sentence and why it could be useful to still send ACK frames. So I propose to further add a sentence like this:

`For example, ACK frames might be preferred by the sender as long as only the initial path with Path ID 0 is in use.`

Reply:
```suggestion
Endpoints MUST process ACK frames that acknowledge 0-RTT packets or 1-RTT packets
for Path ID 0, even after successful negotiation of this extension. For example,
ACK frames might be preferred by the sender as long as only the initial path
with Path ID 0 is in use.
```